aci/containers.go

```diff
@@ -235,12 +235,15 @@ func (cs *aciContainerService) Delete(ctx context.Context, containerID string, r
 
 func (cs *aciContainerService) Inspect(ctx context.Context, containerID string) (containers.Container, error) {
 	groupName, containerName := getGroupAndContainerName(containerID)
+	if containerID == "" {
+		return containers.Container{}, errors.New("cannot inspect empty container ID")
+	}
 
 	cg, err := getACIContainerGroup(ctx, cs.ctx, groupName)
 	if err != nil {
 		return containers.Container{}, err
 	}
-	if cg.StatusCode == http.StatusNoContent {
+	if cg.StatusCode == http.StatusNoContent || cg.Containers == nil {
```

Comment:
Here actually `*http.Response` can be `nil`. `autorest.Response` has a helper function for it. Also `*ContainerGroupProperties`:
```suggestion
	if cg.IsHTTPStatus(http.StatusNoContent) || cg.ContainerGroupProperties == nil || cg.Containers == nil {
```
Actually there is the same case in other parts of the file